winter/django.py

```diff
@@ -84,11 +84,19 @@ def dispatch(winter_view, request: Request, **path_variables):
 def _call_controller_method(controller, route: Route, request: Request):
     method = route.method
     arguments = arguments_resolver.resolve_arguments(route.method, request)
+    handlers_by_exception = get_throws(method)
     try:
         result = method(controller, **arguments)
         return convert_result_to_http_response(request, result, route.method)
-    except tuple(get_throws(method)) as e:
-        result = exceptions_handler.handle(request, e)
+    except tuple(handlers_by_exception) as exception:
+        exception_cls = type(exception)
+        handler = handlers_by_exception.get(exception_cls)
+
+        if handler is not None:
+            result = handler.handle(request, exception)
+            result = convert_result_to_http_response(request, result, handler.handle_method)
+        else:
+            result = exceptions_handler.handle(request, exception)
```

Comment:
Quite a huge amount of code appeared here. `_call_controller_method` is no longer calling method, it's busy mostly with exception handling. Try to encapsulate all the logic in exceptions_handler.

Reply:
I think you can create `ExceptionsHandler` instance for each method. It can be used to handle both overrides and fallback to default handler if needed.
```python
def _create_exception_handler_for_method(method) -> ExceptionHandler:
    method_exception_handler = ExceptionsHandler()
    for exception_cls, handler in get_throws(method):
        method_exception_handler .add_handler(exception_cls, handler)
    method_exception_handler.add_handler(exceptions_handler)
```